### PR TITLE
Add execute task api for MLClient

### PR DESCRIPTION
**Pull request by jackiehanyang**
### Description
Add execute task api for MLClient
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java [hunk 1]
```diff
@@ -182,6 +186,18 @@ public void registerModelGroup(
...
     }
 
+    /**
+     * Execute metrics correlation algorithm
+     * @param name Function name
+     * @param metricsCorrelationInput metrics correlation input
+     * @param listener a listener to be notified of the result
+     */
+    @Override
+    public void executeAlgorithm(FunctionName name, Input metricsCorrelationInput, ActionListener<MLExecuteTaskResponse> listener) {
```
Comment (dhrubo-os) [inline client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java:196]: From here it seems like we are opening up `execute` api only for metrics correlation algorithm? Should we open up `execute api` in general (the way we open up train/predict api for all algorithms)? May be not tied to only metrics correlation algorithm?

@ylwu-amzn what do you think?

Comment (jackiehanyang) [reply client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java:196]: There's no execute api in general. Already synced up with yaliang on this solution offline.

Comment (dhrubo-os) [reply client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java:196]: I see, if that's the case can we add more comment and TODO in the code?

Comment (ylwu-amzn) [reply client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java:196]: @jackiehanyang I think we have some miscommunication here. We only have metrics correlation now, but you should not name the parameter name as `metricsCorrelationInput` here, that's too specific. Suggest change to a more general name like `input` 
```suggestion
    public void executeAlgorithm(FunctionName name, Input input, ActionListener<MLExecuteTaskResponse> listener) {
```

Comment (jackiehanyang) [reply client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java:196]: updated the parameter name here - https://github.com/opensearch-project/ml-commons/pull/1540/files

**Pull request closed without merging**